tests/s25Main/UI/testSmartBitmap.cpp

```diff
@@ -26,6 +30,77 @@ static std::ostream& boost_test_print_type(std::ostream& os, const ColorBGRA col
 }
 } // namespace libsiedler2
 
+namespace rttrOglMock2 {
+RTTR_IGNORE_DIAGNOSTIC("-Wmissing-declarations")
+
+const Point<float>* texturePointer;
+const Point<float>* vertexPointer;
+std::vector<Point<float>> textureCoords;
+std::vector<Point<float>> vertexCoords;
+int callCount;
+
+void APIENTRY glVertexPointer(GLint size, GLenum type, GLsizei stride, const void* pointer)
+{
+    ++callCount;
+    RTTR_Assert(type == GL_FLOAT);
+    RTTR_Assert(stride == 0);
+    RTTR_Assert(size == 2);
+    vertexPointer = static_cast<const Point<float>*>(pointer);
```

Comment:
This is a pointer to a stack variable, isn't it? So you'd need to copy the values.

Reply:
> size is number of floats in point

--> Points is number of float pairs, i.e. size/2

> OpenGL API that it expects the pointer to be valid the moment the glDrawArrays() is called.

Yes but here you have:
```
void foo() {
Points[...] pts
glVertexPointer(pts)
glDraw()
}
void test() {
foo();
//acccess pts;
}
```

I.e. OpenGL gets the valid pointer but you check the pointer after the function has returned and the pointer is no longer valid.

Reply:
Yes, but...

My code assumes that both `glVertexPointer` and `glDrawArrays` are done back to back and in the latter call the points from `texturePointer` and `vertexPointer` are copied into the `textureCoords` and `vertexCoords` for later access. So it's more like:

```cpp
void foo() {
Points[...] pts;
glVertexPointer(pts); // store pts ptr for glDraw()
glDraw(); // copy pts contents to textureCoords/vertexCoords
}
void test() {
foo();
//access textureCoords/vertexCoords
}
```

What am I missing here?